### ISSUE TITLE
Adding JS-Authenticator from jar to different step/flow causes duplicate key value violation

### ISSUE THREAD

**Issue by ckaspro** [keycloak/keycloak#22213]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/ui

### Describe the bug

With the scripts/ feature enabled, adding a JS- Authenticator to a flow works for the first time. After this, when trying to add the same script to another step in the same (sub)-flow, or a new flow, the action is aborted upon saving:
- GUI: Could not update flow: unknown_error
- Logs:
```
WARN  [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-54) SQL Error: 0, SQLState: 23505
ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-54) ERROR: duplicate key value violates unique constraint "constraint_auth_pk"
  Detail: Key (id)=(script-auth_test.js) already exists.
INFO  [org.hibernate.engine.jdbc.batch.internal.AbstractBatchImpl] (executor-thread-54) HHH000010: On release of batch it still contained JDBC statements
```

I already have several flows with the same authenticator script, somehow the upgrade to 21.1.1 worked. Behavior equals to current snapshot.

### Version

21.1.1

### Expected behavior

Allow one script to be added multiple times in one flow / separate flows

### Actual behavior

One script can only be added one time in all of the flows.

### How to Reproduce?

- enable scripts feature
- create and upload authenticator script
- add the script to one sub- flow
- add the same script to another sub- flow or another flow

### Anything else?

Upon searching for solutions, the following issues popped up:
- https://github.com/keycloak/keycloak/issues/21745
- https://github.com/keycloak/keycloak/issues/20299

**Comment by ckaspro** [comment]
I forgot to mention: it is also impossible to remove- and then re- add a JS- Authenticator:
- remove JS- Authenticator
- try adding the same JS- Authenticator back in, or at another stage/flow
-> results in the same error as above.
That means, if one needs to change a flow, one needs to upload the authenticator with a new file- name, otherwise it will always me rejected.
I fear this might also clutter the database quite a bit.

**Comment by edewit** [comment]
@ckaspro This isn't a UI issue right

**Comment by ckaspro** [comment]
> @ckaspro This isn't a UI issue right

No, this DB issue is affecting the usage and handling of authenticators...

**Comment by pedroigor** [comment]
@ckaspro I'm able to reproduce the error in `21.1.1` (and `21.1.2`). Would mind trying it out as the nightly image?

The fix is this https://github.com/keycloak/keycloak/pull/22031 and it was backported to the 22. It should be available in 22.0.2.

**Comment by ckaspro** [comment]
> @ckaspro I'm able to reproduce the error in `21.1.1` (and `21.1.2`). Would mind trying it out the nightly image?
> 
> The fix is this #22129 and it was backported to the 22. It should be available in 22.0.2, once released.

The current nightly fixes the issue - thanks you!
